File: 백준/Silver/20044. Project Teams/Project Teams.py
# 10000개의 정수가 주어지는데 최소를 결국 찾아야함
# 끝과 끝을 더해서 가장 최소를 찾아야함

import sys

N = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))

arr.sort()

# 최솟값
min_value = 200000

for i in range(2 * N):
    if i == (2 * N) - 1:
        break

    if min_value > arr[i] + arr[(2 * N) - i - 1]:
        min_value = arr[i] + arr[(2 * N) - i - 1]

print(min_value)

File: 백준/Silver/20044. Project Teams/Project Teams.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {


    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bf.readLine()) * 2;
        int[] arr = new int[n];

        String[] sa = bf.readLine().split(" ");
        for (int i = 0; i < n; i++)
            arr[i] = Integer.parseInt(sa[i]);

        Arrays.sort(arr);

        int min = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            int sum = arr[i] + arr[n - i - 1];
            if (min > sum)
                min = sum;
        }

        System.out.println(min);

    }
}
